forgotten-sender/listing-02/main.go

```diff
@@ -0,0 +1,56 @@
+// Example program showing a goroutine leak. We launch a goroutine that sends
+// on a channel but sometimes there is no other goroutine available to receive.
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"runtime"
+	"time"
+)
+
+func main() {
+
+	// Seed the random number generator so we get different results.
+	// The seed 42 always results in a timeout and thus a leak.
+	// The seed 99 always results in success and thus no leak.
+	// In production we may seed using the current time like
+	// rand.Seed(time.Now().UnixNano())
+	rand.Seed(42)
+
+	fmt.Printf("Number of goroutines: %d\n\n", runtime.NumGoroutine())
+
+	process()
+
+	fmt.Printf("\nNumber of goroutines: %d\n", runtime.NumGoroutine())
+}
+
+func process() {
+
+	// Make a channel for our goroutine to report its result.
+	ch := make(chan string)
+
+	// Start a worker to do some work then send on the channel.
+	go func() {
+		ch <- doSomeWork()
+	}()
+
+	// Create a timeout channel. In 100ms a value will be sent on this channel.
```

Comment:
// Create a timeout channel. In 100ms a Time value will be sent on this channel.